homeassistant/components/input_text/__init__.py

```diff
@@ -110,52 +180,53 @@ def _cv_input_text(cfg):
         SERVICE_SET_VALUE, {vol.Required(ATTR_VALUE): cv.string}, "async_set_value"
     )
 
-    if entities:
-        await component.async_add_entities(entities)
     return True
 
 
-async def _async_process_config(config):
-    """Process config and create list of entities."""
-    entities = []
-
-    for object_id, cfg in config[DOMAIN].items():
-        if cfg is None:
-            cfg = {}
-        name = cfg.get(CONF_NAME)
-        minimum = cfg.get(CONF_MIN, CONF_MIN_VALUE)
-        maximum = cfg.get(CONF_MAX, CONF_MAX_VALUE)
-        initial = cfg.get(CONF_INITIAL)
-        icon = cfg.get(CONF_ICON)
-        unit = cfg.get(ATTR_UNIT_OF_MEASUREMENT)
-        pattern = cfg.get(ATTR_PATTERN)
-        mode = cfg.get(CONF_MODE)
-
-        entities.append(
-            InputText(
-                object_id, name, initial, minimum, maximum, icon, unit, pattern, mode
-            )
-        )
+class InputTextStorageCollection(collection.StorageCollection):
+    """Input storage based collection."""
+
+    CREATE_SCHEMA = vol.Schema(vol.All(CREATE_FIELDS, _cv_input_text))
+    UPDATE_SCHEMA = vol.Schema(UPDATE_FIELDS)
+
+    async def _process_create_data(self, data: typing.Dict) -> typing.Dict:
+        """Validate the config is valid."""
+        return self.CREATE_SCHEMA(data)
 
-    return entities
+    @callback
+    def _get_suggested_id(self, info: typing.Dict) -> str:
+        """Suggest an ID based on the config."""
+        return info[CONF_NAME]
```

Comment:
Shouldn't this use the optional `CONF_ID` field that you defined in `CREATEA_FIELDS` ?

Reply:
I take it my [question](https://github.com/home-assistant/home-assistant/pull/30576#discussion_r364070821) is still in the back log :)

a) I wanted `CONF_ID` to be optional and I have it there only for `{**data, **update_data}` validation
b) can `_get_suggested_id()` return None? Or is the idea to return optional `CONF_ID` if present and if not then return `CONF_NAME`?

Reply:
Optional conf_id or else name.

Reply:
The idea is that we to move away from UUID because people don't like it when those end up in configs.

Reply:
I'll remove `CONF_ID` from the schema and just going to use `CONF_NAME`  so no UUIDs in the configs. I concur with that.